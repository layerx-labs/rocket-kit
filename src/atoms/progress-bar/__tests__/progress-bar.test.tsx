import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { ProgressBar } from '../../..';

describe('ProgressBar', () => {
  it('renders', () => {
    const { asFragment } = render(<ProgressBar progress={10} value="10%" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const progressText = '10%';
    render(<ProgressBar progress={10} value={progressText} />);
    screen.getByText(progressText);
  });
});
