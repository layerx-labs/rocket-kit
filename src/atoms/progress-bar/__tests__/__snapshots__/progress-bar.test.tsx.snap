// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 999px;
  background-color: #e0e0e0;
  height: 0.625rem;
}

.c2 {
  border-radius: 999px;
  background-color: hsl(186,62%,59%);
  height: 100%;
  width: 10%;
  -webkit-transition-duration: 2s;
  transition-duration: 2s;
}

.c3 {
  margin-left: 0.3125rem;
  font-size: 0.75rem;
  font-weight: 700;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      10%
    </div>
  </div>
</DocumentFragment>
`;
